stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt

```diff
@@ -182,7 +182,13 @@ internal class ChatDomainImpl internal constructor(
 
     @VisibleForTesting
     val defaultConfig: Config = Config(connectEventsEnabled = true, muteEnabled = true)
-    internal var repos: RepositoryFacade = createNoOpRepos()
+    internal var repos: RepositoryFacade
```

Comment:
Could we add some docs here to make this change more visible for the future.

Otherwise - this makes absolute sense to me, getting an instance lazily is usually the way to go!

Reply:
Why don't we create a "real NoOp"  instance?
The problem here is that we are calling "NoOp" to an implementation that use the inMemory DB.
We can create a `RepositoryFacadeNoOp` class that does nothing, then we don't need to create the InMemoryDB.

Reply:
@JcMinarro That would be super complex. We would need to create a NoOp version of all our DAO, also `ChatDatabase` extends `RoomDatabase` which should be created using `databaseBuilder` or `inMemoryDatabaseBuilder`. Take a look at its documentation for `RoomDatabase`:

> Creates a RoomDatabase.
You cannot create an instance of a database, instead, you should acquire it via Room.databaseBuilder(Context, Class, String) or Room.inMemoryDatabaseBuilder(Context, Class).

One thing that we could do is not create an in-memory database if the user is not set and simply return an error instead to avoid that customers initialize database twice without intention.

Reply:
Ok, I don't want to block this PR more because of it.
We can iterate over it later.
BTW, Could you add some docs about why it is done that way to be sure it is not refactored (and reintroducing the bug again) in the future?

Reply:
Done